Support automatic real nesting of comments without placeholders, if supported by the language.

klaasb opened the pull request:
E.g. for haskell, scheme, scala, lisp, ocaml, sml. Fixes issue #131

Implemented by adding 'nested'/'nestedAlt' keys to the language delimiter defintions.

alerque commented:
Hey @klaasb and thanks for taking the time to contribute. I've recently been added as a maintainer on this project and am working my way through the backlog of issues and contributions. I'm having some trouble reviewing this one and was wondering if I could get some help. For one thing with all the other merges I've done there are now some merge conflicts in this branch. Can you rebase it to take care of those?

Also I'm having trouble duplicating the issue that this would solve (#131). The exact procedure explained there behaves correctly for me (by changing the nested comment to an alternate syntax) so I was wondering if there is even still a need for this. If it's a better fix that whatever has already been done I'd love to consider it, but can you at least review it and comment on whether and why it is still needed?

klaasb commented:
Fixed the merge commits, rebased against master and added nesting attributes for Racket, as I saw that scrolling past. However, more languages with nesting might have been added since I wrote this PR.

alerque commented:
Thanks a bunch @klaasb. This seems to work in my testing so I've gone ahead and merged it. I guess now we wait for people who actually use nested languages to tell us if we broke something. Thanks for the help.
